### DISCUSSION TITLE
Why does SvelteFlowProvider delete all nodes on destroy?

### DISCUSSION THREAD

**Discussion by darabos** [xyflow/xyflow#4179] [General]
SvelteFlowProvider calls `reset()` on destroy:
https://github.com/xyflow/xyflow/blob/reactflow-11.10.3.tgz/packages/svelte/src/lib/components/SvelteFlowProvider/SvelteFlowProvider.svelte#L28

`reset()` deletes nodes and edges:
https://github.com/xyflow/xyflow/blob/reactflow-11.10.3.tgz/packages/svelte/src/lib/store/index.ts#L338

I subscribe to the nodes and edges stores to persist edits on the backend. When the editor is closed, this `reset()` call triggers a final update that wipes out everything on the backend.

I think maybe this is accidental? At the time when the `reset()` call was added to SvelteFlowProvider, `reset()` wasn't clearing `nodes` and `edges`. That was added in https://github.com/xyflow/xyflow/pull/3138/commits/d53b7d21c0b1d6a481a92cc27ce31eff2826903c. It's part of a bigger PR, so the reason is not clear. It may be better to remove these two lines. `nodes` and `edges` are user-controlled stores. Let the user decide if they should be reset or not.

**Comment by moklick** [comment]
Thanks, we will check it!
